### PR TITLE
[lldb] Implement TypeSystemSwiftTypeRef::GetNumFields

### PR DESCRIPTION
**Pull request by kastiglione**
Implement `TypeSystemSwiftTypeRef::GetNumFields`.

rdar://68171349

### CODE REVIEW

**Comment by kastiglione** [comment]
@swift-ci test

File: lldb/source/Target/SwiftLanguageRuntimeDynamicTypeResolution.cpp [hunk 1]
```diff
@@ -1093,6 +1093,68 @@ SwiftLanguageRuntimeImpl::GetNumChildren(CompilerType type,
...
+    case ReferenceKind::Unmanaged:
+      if (auto referent = GetWeakReferent(*ts, type))
+        return referent.GetNumFields(exe_ctx);
+      return 0;
+    case ReferenceKind::Strong:
+      TypeConverter tc(GetReflectionContext()->getBuilder());
+      LLDBTypeInfoProvider tip(*this, *ts);
+      auto *cti = tc.getClassInstanceTypeInfo(tr, 0, &tip);
+      if (auto *rti = llvm::dyn_cast_or_null<RecordTypeInfo>(cti)) {
+        return rti->getNumFields();
```
Comment (kastiglione) [inline lldb/source/Target/SwiftLanguageRuntimeDynamicTypeResolution.cpp:1148]: only reference types differ from `GetNumChildren`, so all the other kinds could be extracted to a shared implementation.

**Comment by kastiglione** [comment]
Marked as draft because there may be more `GetNumFields` callsites that need to pass an execution context.

**Comment by adrian-prantl** [comment]
This is looking good.

**Comment by kastiglione** [comment]
@swift-ci test

**Review by adrian-prantl — APPROVED** [review]

**Comment by kastiglione** [comment]
@swift-ci test macOS Platform

**Pull request merged**